nvm.sh

```diff
@@ -2317,6 +2386,55 @@ nvm_npmrc_bad_news_bears() {
   return 1
 }
 
+nvm_windows_drive_posix() {
+  local location
+  location=$(nvm_grep -iEe "^$1 " /etc/mtab 2>/dev/null | command awk '{ print $2 }')
+  # Cygwin seems not to have /etc/mtab
```

Comment:
Cygwin may not have `/etc/mtab` (and judging from some internet complaints, WSL might not always have it either), but AFAICT all three alternatives have `/proc/mounts`, with `/etc/mtab` just being a symlink when it exists.

Also, for both Cygwin and MSYS, the `cygpath` utility exists, but also `$(cd drive:/ ; pwd)` will do the trick since both Cygwin and MSYS support using Windows paths directly in at least bash and sh (that I tried).  I don't think WSL will handle that, though.  So probably, using /proc/mounts will do the trick for all three.

(Off topic: thanks for the work on this!  I've been using nvm on Cygwin for a very long time (5+ years), manually installing the binaries and tweaking the config files by hand to make it work.  It'll be nice not to have to do that any more!)

Reply:
Also, apparently WSL has a [`wslpath` utility](https://github.com/MicrosoftDocs/WSL/releases/tag/17046) that supports `-u`, `-a`, `-m` and `-w` with identical functionality to the same short options on `cygpath`.  So using cygpath or wslpath according to platform would allow you to replace most of `nvm_ensure_posix_path` with `location=$(cygpath_or_wslpath -u "$location")`, and have the additional advantage of properly translating non-drive mount points as well.

Reply:
Thanks for those inputs, I ll check that ;)